Source/ARTRest.m

```diff
@@ -530,6 +530,9 @@ BOOL ARTstartHandlingUncaughtExceptions(ARTRest *self) {
 }
 
 void ARTstopHandlingUncaughtExceptions(ARTRest *self) {
+    if (!self) {
```

Comment:
@tcard Another thing. I was facing a deadlock as well. I spent some time at my lunch time to figure it out but the time wasn't enough. I will open a new PR.